app/healthcheck.go

```diff
@@ -0,0 +1,72 @@
+package app
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/types/rest"
+	"github.com/gorilla/mux"
+)
+
+var (
+	LatestBlockHeight int64
+	lastResult        bool
+)
+
+const progressCheckPeriod = 1 * time.Minute
+
+func HealthcheckRegister(clientCtx client.Context, r *mux.Router) {
+	go RunHealthCheck(clientCtx)
+	r.HandleFunc("/healthcheck", HealthcheckRequestHandlerFn(clientCtx)).Methods("GET")
+}
+
+func getLatestHeight(clientCtx client.Context) (int64, error) {
```

Comment:
FYI, i've took this from the `/block` API implementatiob

Reply:
Invariants can be checked manually via [messages](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/02-messages-and-queries.md), but most often they are checked automatically at the end of each block.
So it can also be used as a trigger. Like updating the time at the beginning of the block and trigger checking if there was no update for some time.
I am good with the two options, up to you.